### DISCUSSION TITLE
Make windows ignore mouse events

### DISCUSSION THREAD

**Discussion by YeShangwei** [AvaloniaUI/Avalonia#8617] [Q&A] [answered]
Hi,
I try to draw selection windows on OSX, and I wonder how to make windows ignore mouse events on OSX. 
I found I can set _setIgnoreMouseEvents_ in objective-c. Is there any method in Avalonia?

**Comment by kekekeks** [comment, marked as answer]
`((IMacOSTopLevelPlatformHandle)Window.PlatformImpl).NSWindow` to get `NSWindow` handle and then use MonoMac to call ObjC APIs

**Reply by YeShangwei** [reply]
Thank you,
but I got an error:
**Unable to cast object of type 'Avalonia.Native.WindowImpl' to type 'Avalonia.Platform.IMacOSTopLevelPlatformHandle'**

**Reply by maxkatz6** [reply]
You need to cast Window.PlatformImpl.Handle

**Reply by YeShangwei** [reply]
Thank you,
when try to handle this window, I get a error:
**AppKit Consistency error: you are calling a method that can only be invoked from the UI thread.**

My code is here:
```
IntPtr handler = ((IMacOSTopLevelPlatformHandle)temp.PlatformImpl.Handle).NSWindow;
NSWindow nsWindow = new NSWindow(handler);
nsWindow.IgnoresMouseEvents = true;
```

**Reply by timunie** [reply]
You maybe need to use `Dispatcher`, see: https://docs.avaloniaui.net/guides/basics/accessing-the-ui-thread

**Reply by YeShangwei** [reply]
Thank you, I try it, but it seems not work. And there's something interesting: `Dispatcher.UIThread.CheckAccess()`  returns `true`, but error message is: 
**AppKit Consistency error: you are calling a method that can only be invoked from the UI thread.**

**Reply by maxkatz6** [reply]
You probably are missing `NSApplication.Init();` in the beginning of your app. It should be called only on macOS before every other call to macOS api.
